tests/Add-DbaAgDatabase.Tests.ps1

```diff
@@ -38,49 +47,130 @@ Describe "Add-DbaAgDatabase" -Tag "UnitTests" {
+        $filesToRemove += $backup.Path
+
+        $null = New-DbaDatabase -SqlInstance $TestConfig.instance3 -Name $existingDbWithoutBackup
+
+        # We want to run all commands outside of the BeforeAll block without EnableException to be able to test for specific warnings.
+        $PSDefaultParameterValues.Remove('*-Dba*:EnableException')
     }
 
     AfterAll {
-        $null = Remove-DbaAvailabilityGroup -SqlInstance $server -AvailabilityGroup $agname -Confirm:$false
-        $null = Remove-DbaDatabase -SqlInstance $server -Database $dbname, $newdbname -Confirm:$false
+        # We want to run all commands in the AfterAll block with EnableException to ensure that the test fails if the cleanup fails.
+        $PSDefaultParameterValues['*-Dba*:EnableException'] = $true
+
+        $null = Remove-DbaAvailabilityGroup -SqlInstance $TestConfig.instance3 -AvailabilityGroup $agName
+        $null = Get-DbaEndpoint -SqlInstance $TestConfig.instance3 -Type DatabaseMirroring | Remove-DbaEndpoint
+        $null = Remove-DbaDatabase -SqlInstance $TestConfig.instance3 -Database $existingDbWithBackup, $existingDbWithoutBackup
+        Remove-Item -Path $filesToRemove
+
+        # As this is the last block we do not need to rest the $PSDefaultParameterValues.
     }
 
     Context "When adding AG database" {
+        # We use the BeforeAll to run the test itself.
+        # Results are saved in $results.
         BeforeAll {
-            $server.Query("create database $newdbname")
-            $backup = Get-DbaDatabase -SqlInstance $TestConfig.instance3 -Database $newdbname | Backup-DbaDatabase
             $splatAddAgDb = @{
-                SqlInstance = $TestConfig.instance3
-                AvailabilityGroup = $agname
-                Database = $newdbname
-                Confirm = $false
+                SqlInstance       = $TestConfig.instance3
+                AvailabilityGroup = $agName
+                Database          = $existingDbWithBackup
             }
             $results = Add-DbaAgDatabase @splatAddAgDb
         }
 
+        It "Does not warn" {
+            $WarnVar | Should -BeNullOrEmpty
+        }
+
         It "Returns proper results" {
-            $results.AvailabilityGroup | Should -Be $agname
-            $results.Name | Should -Be $newdbname
+            $results.AvailabilityGroup | Should -Be $agName
+            $results.Name | Should -Be $existingDbWithBackup
             $results.IsJoined | Should -Be $true
         }
     }
-} #$TestConfig.instance2 for appveyor
+
+    Context "When adding AG database that does not have a backup" {
+        BeforeAll {
+            $splatAddAgDb = @{
+                SqlInstance       = $TestConfig.instance3
+                AvailabilityGroup = $agName
+                Database          = $existingDbWithoutBackup
+                # As we don't want an output, we suppress the warning.
+                # But we can still test the warning because WarningVariable is set globally.
```

Comment:
What do you think about setting a WarningVariable globally? But we have to agree on the name. I sometimes use just `warn`, but I think `WarnVar` as a short version of `WarningVariable` is a good option.

Reply:
sounds goodi like $warn but happy to have w/e standardaized